src/v2/modules/Duration.mjs

```diff
@@ -0,0 +1,139 @@
+// -----------------------------------------------------------------------------
+// Copyright 2022 Open Climate Tech Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+/**
+ * Provides static constants and methods for dealing with lengths of time.
+ */
+export default class Duration {
+  static MILLISECOND = 1
+  static SECOND = 1000 * Duration.MILLISECOND
+  static MINUTE = 60 * Duration.SECOND
+  static HOUR = 60 * Duration.MINUTE
+  static DAY = 24 * Duration.HOUR
+  static WEEK = 7 * Duration.DAY
+
+  /**
+   * The discrete, whole components of the duration between a start date-time
+   * and an end date-time.
+   *
+   * @typedef {Object} Distance
+   * @property {number} years - The total whole successive calendar years.
+   * @property {number} months - The remaining whole successive calendar months.
+   * @property {number} weeks - The remaining whole 7-day periods.
+   * @property {number} days - The remaining whole 24-hour days.
+   * @property {number} hours - The remaining whole hours.
+   * @property {number} minutes - The remaining whole minutes.
+   * @property {number} seconds - The remaining whole seconds.
+   * @property {number} milliseconds - The remaining milliseconds.
+   */
+
+  /**
+   * Calculates the difference between timestamps as a person using a calendar
+   * might (including hours gained or lost to local daylight saving time).
+   *
+   *   - 1 month is the time between the same date in successive calendar months
+   *     such as Feb 2 - Mar 2.
+   *   - 1 year is the time between the same date in successive calendar years
+   *     such as Feb 2, 2022 - Feb 2, 2023.
+   *
+   * @param {Date} startDate - The start timestamp for comparison.
+   * @param {Date} endDate - The end timestamp for comparison.
+   *
+   * @returns {Distance}
+   */
+  static calculateCalendarDistance(startDate, endDate) {
```

Comment:
Fair enough. This is accurate about time/distance in a way that is compatible with how how people naturally think about it. It’s also _extremely_ minimal compared to packages like moment.js or even date-fns. And it’s already done and tested. If you want to drop it because it’s more code than you want to maintain, or covering distances more than we’re likely to see in practice seems like total overkill, or there’s a material performance difference—that’s not a problem.